plugins/modules/files/json_file.py

```diff
@@ -0,0 +1,540 @@
+#        ],
+#        "F": 7
+#    }
+# }
+
+- name: "Ensure values are present in the json diff, also check file format."
+  community.general.json_file:
+    path: "/my/dest/file.json"
+    indent: 2
+    diff_on_value: false
+    value:
+      A: 1
+    state: "present"
+# There is not diff in the values but the file format is changed (intent 4
+# There is a change.
+# Get the following content in /my/dest/file.json :
+to 2).
+#{
+#  "A": 1,
+#  "B": 2,
+#  "C": {
+#    "D": 3,
+#    "E": [
+#      4,
+#      5,
+#      6
+#    ]
+#  }
+#}
+'''
+
+RETURN = r'''
+  changed:
+    description:
+      - Indicate whether the json file is changed or not.
+    returned: always
+    type: bool
+  created:
+    description:
+      - Indicate if the file json is created or not.
+    returned: success
+    type: bool
+  diff:
+    description:
+      - Contain a dict with two elements, C(after) contain the value after
+        the change and C(before) contain values before the change.
+    returned: changed
+    type: dict
+  failed:
+    description:
+      - Indicate if the module has failed or not.
+    returned: always
+    type: bool
+  result:
+    description:
+      - Contain the values that are finally present in the json file.
+    returned: success
+    type: dict
+'''
+
+
+class JsonFile(DestFileModuleHelper):
+
+    module = dict(
+        argument_spec=dict(
+            path=dict(type='path', required=True, aliases=['dest']),
+            state=dict(
+                type='str',
+                default='present',
+                choices=[
+                        'present',
+                        'absent',
+                        'identic',
+                ]
+            ),
+            list_diff_type=dict(
+                type='str',
+                default='value',
+                choices=[
+                        'value',
+                        'index',
+                ],
+            ),
+            value=dict(type=check_type_dict_or_list, required=True),
+            create=dict(type='bool', default=False),
+            diff_on_value=dict(type='bool', default=True),
+            backup=dict(type='bool', default=False),
+            indent=dict(type='int', default=4),
+            sort_keys=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+    )
+
+    def __load_result_data__(self) -> dict:
+        content = []
+        try:
+            with open(self.vars.path, 'r') as file:
+                content = file.readlines()
+        except FileNotFoundError:
+            pass
+        except json.JSONDecodeError as ex:
+            raise ModuleHelperException(
+                msg=f'Failed to decode json in {self.vars["path"]}',
+                exception=ex)
+        if ''.join(content).strip() == '':
+            content = ['{}']
+        if self.vars['diff_on_value']:
+            self.vars.set(self.var_result_data,
+                          json.loads(''.join(content)), diff=True)
```

Comment:
Please avoid breaking lines if not necessary. We allow up to 160 chars in a line, but even if you think that's too much, this combined line doesn't even have 90 characters.
```suggestion
            self.vars.set(self.var_result_data, json.loads(''.join(content)), diff=True)
```